packages/ai-jsx/src/batteries/docs.tsx

```diff
@@ -1,154 +1,498 @@
- *  * Do a k-nearest-neighbors search in a vector store.
+ *  Examples:
+ *      Simple handful of documents:
+ *          The response to the initial request contains only a page of
+ *          documents. This could include a next_page_token for more
+ *          documents in the single default partition if needed.
+ *      Web crawl:
+ *          Each URL corresponds to a partition and the responses never include
+ *          tokens. The initial response only includes partitions, one for each
+ *          root URL to crawl. Each subsequent request includes the partition
+ *          (the URL) and the corresponding response contains a page with a
+ *          single document - the resource at that URL. If the document links
+ *          to other resources that should be included in the corpus, the
+ *          response also contains those URLs as new partitions. The process
+ *          repeats for all partitions until there are no known incomplete
+ *          partitions or until crawl limits are reached.
+ *      Database:
+ *          Consider a database with a parent table keyed by parent_id and an
+ *          interleaved child table keyed by (parent_id, child_id) whose rows
+ *          correspond to corpus documents. This loader will use tokens that
+ *          encode a read timestamp (for consistency) and an offset to be used
+ *          in combination with a static page size.
  *
- * Once you have the Documents, you may wish to do some combination of:
- *  1. Inserting them directly into a prompt
- *  2. Adding them to a vector store.
- *  3. Chunking them into smaller documents (use a `Chunk` method).
+ *          Upon receiving the initial request, the loader chooses a commit
+ *          timestamp to use for all reads and returns a partition for each
+ *          parent_id along with a first page token indicating the chosen read
+ *          timestamp and an offset of 0.
  *
- * In LangChain, there are loaders and retrievers. In AI.JSX, there are only loaders. Retrievers are a special
- * case of loaders.
+ *          For each partition, the loader then receives requests with the
+ *          partition (a parent_id) and a page token (the read timestamp and
+ *          offest). It responds with documents corresponding to the next page
+ *          size child rows within the given parent. If more children exist,
+ *          the response includes a next page token with the same read
+ *          timestamp and an incremented offset. This repeats until there are
+ *          no more children, at which point the response has no
+ *          next page token and the partition is complete.
+ *
+ *          Note: Including multiple parent_ids in each partition would also
+ *              work and would be an effective way to limit parallelism if
+ *              desired.
  */
-// TODO: maybe rename to DocumentProvider
-export type Loader = (...args: any[]) => Promise<Document[]>;
+export namespace CorpusLoading {
```

Comment:
Hmm, why do we need to use a namespace?

Reply:
I thought it was a useful way to group related types, but this is something I was hoping to get feedback on.

Reply:
I think we probably don't need `namespace`, and can just rely on the file itself as a way to group types. We have a bunch of types in `src/index.ts`, and they're not super tightly grouped, but it's been serving us fine.